patches/minecraft/net/minecraft/pathfinding/WalkNodeProcessor.java.patch

```diff
@@ -0,0 +1,18 @@
+--- ../src-base/minecraft/net/minecraft/pathfinding/WalkNodeProcessor.java
++++ ../src-work/minecraft/net/minecraft/pathfinding/WalkNodeProcessor.java
+@@ -415,6 +415,7 @@
+                         {
+                             pathnodetype = PathNodeType.DANGER_FIRE;
+                         }
++                        else if(block1.isBurning(p_186330_1_,blockpos$pooledmutableblockpos.func_181079_c(j +p_186330_2_, p_186330_3_, i + p_186330_4_))) pathnodetype = PathNodeType.DAMAGE_FIRE;
+                     }
+                 }
+             }
+@@ -430,6 +431,7 @@
+         IBlockState iblockstate = p_189553_1_.func_180495_p(blockpos);
+         Block block = iblockstate.func_177230_c();
+         Material material = iblockstate.func_185904_a();
++        if(block.isBurning(p_189553_1_, blockpos)) block = Blocks.field_150480_ab; // If the block is burning, make the AI think it is a fire block
```

Comment:
That.. would actually be better as it is more clear and doesn't rely on the existing code to stay the same.